base/linalg/blas.jl

```diff
@@ -112,6 +112,29 @@ function set_num_threads(n::Integer)
     return nothing
 end
 
+"""
+    get_num_threads(n)
+
+Get the number of threads the BLAS library can use.
+"""
+function get_num_threads()
```

Comment:
@eschnett you tried to do this in #10028 / #15008 - what was the issue you had in the PR?

It may also be worth making this conditional on openblas version, since I don't think they've had this API for all that long.